Lib/test/test_pathlib/test_pathlib_abc.py

```diff
@@ -2072,6 +2072,125 @@ def test_copy_dangling_symlink(self):
         self.assertTrue(target2.joinpath('link').is_symlink())
         self.assertEqual(target2.joinpath('link').readlink(), self.cls('nonexistent'))
 
+    def test_move_file(self):
+        base = self.cls(self.base)
+        source = base / 'fileA'
+        source_text = source.read_text()
+        target = base / 'fileA_moved'
+        result = source.move(target)
+        self.assertEqual(result, target)
+        self.assertFalse(source.exists())
+        self.assertTrue(target.exists())
+        self.assertEqual(source_text, target.read_text())
+
+    def test_move_file_to_file(self):
+        base = self.cls(self.base)
+        source = base / 'fileA'
+        source_text = source.read_text()
+        target = base / 'dirB' / 'fileB'
+        result = source.move(target)
+        self.assertEqual(result, target)
+        self.assertFalse(source.exists())
+        self.assertTrue(target.exists())
+        self.assertEqual(source_text, target.read_text())
+
+    def test_move_file_to_dir(self):
+        base = self.cls(self.base)
+        source = base / 'fileA'
+        target = base / 'dirB'
```

Comment:
Did we actually cover the case "move file to a different empty directory"? AFAIU, this wouldn't raise an OSError (what we did test was "move file to a file in a different directory"). Namely, did we cover:

```py
source = base / 'fileA'
target = base / 'emptydir'
target.mkdir(exists_ok=False)
source.move(target)
self.assertFalse(source.exists())
# I think the target now becomes a file according to os.replace() docs.
self.assertFalse(target.is_dir())
```

Reply:
> Did we actually cover the case "move file to a different empty directory"? AFAIU, this wouldn't raise an OSError 

The `os.replace()` docs actually don't specify what happens when the destination is an existing empty directory. In practice is varies slightly by OS, e.g. `os.replace(dir, empty_dir)` succeeds on Windows but fails on POSIX. I've been careful in the `move()` docs not to specify this either, as we're at the mercy of whatever `replace()` does. As such I don't think it's worth testing because it's not part of the contract we want to offer users!

Reply:
Oh. I see. Since the docs say that `os.replace` is called, I think it's fine then to leave it as is.